crates/store/src/types/context.rs

```diff
@@ -54,6 +54,12 @@ impl PredefinedEntry for key::ContextState {
     type DataType<'a> = ContextState<'a>;
 }
 
+#[derive(Eq, Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct ContextIdentity {
+    pub public_key: PublicKey,
```

Comment:
this is already in the key used to access this entry, `public_key` is not needed here

Reply:
That's fair, but I was under the impression that you had said for it to be there, and to convert to and from the other type(s). We ended up changing the `ContextIdentity` primitive to `KeyPair`, and as we still have that type used then there's no way to convert the `KeyPair` to and from the `ContextIdentity` without losing data if we remove this field. I did actually remove it and spend some time carefully exploring what the repercussions would be like, but I did not feel comfortable as they deviate too far from what we've previously discussed. Therefore I've kept this field in for now - if you 100% want it removed then please advise what you want done with the `KeyPair`, the interaction between the two types, and the wider usage.

Reply:
> but I was under the impression that you had said for it to be there, and to convert to and from the other type(s).

Not at all, I don't foresee any use for `KeyPair` aside as an input to `create_context` and `join_context` (which could just be parameters at that point).

But if you had to keep a `KeyPair` in, then it's "conversion to the store entry"  would be:

```rust
let mut handle = store.handle();

handle.put(
    &calimero_store::key::ContextIdentity::new(context_id, key_pair.public_key),
    &calimero_store::types::ContextIdentity {
        private_key: keypair.private_key
    }
);
```

it doesn't have to warrant a conversion to an identical structure for storage